patches/@mui+material+5.4.3.patch

```diff
@@ -0,0 +1,44 @@
+diff --git a/node_modules/@mui/material/styles/shadows.d.ts b/node_modules/@mui/material/styles/shadows.d.ts
+index 2f7e309..a53bc36 100644
+--- a/node_modules/@mui/material/styles/shadows.d.ts
++++ b/node_modules/@mui/material/styles/shadows.d.ts
+@@ -1,29 +1,12 @@
+-export type Shadows = [
+-  'none',
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-  string,
+-];
++export type Shadows = {
++  none: string;
++  xs: string;
++  sm: string;
++  md: string;
++  lg: string;
++  xl: string;
++  xxl: string;
```

Comment:
Created a patch since MUI doesn't currently support this format of setting shadows. Although they plan to support in [v6](https://github.com/mui/material-ui/issues/28820#issuecomment-1000280866)

Reply:
If we forcefully override this type, I just want to make sure we're on the same page that this means we should no longer use certain props on some of the components. For instance, for the MUI Paper component we can no longer use the `elevation` prop to set the shadow, as it expects an integer to be an index in the array:
```JSX
<Paper
  elevation={10} // we can no longer use this prop
  sx={{
    boxShadow: (theme) => theme.shadows.xl, // we have to do this instead
  }}
>
   Paper!
</Paper>
```

Reply:
I think that's okay, I just want to make it clear.

Reply:
Thanks for catching @benwerner01. I didn't consider the Paper component when creating the patch. I tested and saw we could actually do this

```
<Paper
  elevation={"lg"} // which I feel is much easier
  sx={{
    boxShadow: "xl", // or do this
  }}
>
   Paper!
</Paper>
```
But to get typing we need to override the types for PaperProps. Didn't have enough time to do that, plus I figured there might be some other components that make use of the previous setup that I haven't accounted for.

I decided to revert the patch and instead add two new keys to our theme `boxShadows` and `dropShadows`, that way we can do 

```
<Paper
  elevation={10} // this
  sx={{
    boxShadow: (theme) => theme.boxShadows.xl, // and this
  }}
>
   Paper!
</Paper>
```

Commit

Reply:
Thanks Teeno, we can always refine this further down the line.